### PR TITLE
[#12] [Backend] As a user, I can see the survey on the Dashboard page

### LINKED ISSUE #12: [Backend] As a user, I can see the survey on the Dashboard page
## Why
When the user sign in, he should see the Dashboard page with the surveys.

## Acceptance Criteria
- Add GET `/api/v1/surveys?page[number]={pageNumber}&page[size]={pageSize}` API
- Write unittest correspondingly.

### PR DESCRIPTION
**Pull request by manh-t**
Close #12

## What happened 👀

- Implement the API GET `api/v1/surveys`
- Create `BaseAuth` for the APIs that needs authentication
- The API returns a list of survey, so I prepare the `deserializableList`.

## Insight 📝

N/A

## Proof Of Work 📹

image

### CODE REVIEW

**Comment by manh-t** [comment]
🚀 Deployed on https://64bf78e959c62834e19da12e--manh-react-survey.netlify.app

**Comment by manh-t** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Coverage is a little low, take a look</td>
    </tr>
  </tbody>
</table>

## Coverage in All Files
File | Line Coverage | Statement Coverage | Function Coverage | Branch Coverage
---- | ------------: | -----------------: | ----------------: | --------------:
[src/../Dashboard/index.tsx](../blob/664783e/src/screens/Dashboard/index.tsx) | (1/6) 17% | (1/6) 17% | (0/4) 0% | (0/0) NaN%
[src/../v1/requestManager.ts](../blob/664783e/src/lib/request/v1/requestManager.ts) | (9/18) 50% | (9/18) 50% | (4/6) 67% | (1/3) 33%
[src/config/index.ts](../blob/664783e/src/config/index.ts) | (5/9) 56% | (5/9) 56% | (0/3) 0% | (0/4) 0%
[src/tests/error.ts](../blob/664783e/src/tests/error.ts) | (2/3) 67% | (2/3) 67% | (1/2) 50% | (1/3) 33%
[src/screens/SignIn/index.tsx](../blob/664783e/src/screens/SignIn/index.tsx) | (12/16) 75% | (12/16) 75% | (3/6) 50% | (6/6) 100%
[src/../Content/index.tsx](../blob/664783e/src/components/Dashboard/Content/index.tsx) | (8/10) 80% | (8/10) 80% | (4/6) 67% | (3/5) 60%
[src/helpers/authentication.ts](../blob/664783e/src/helpers/authentication.ts) | (10/11) 91% | (10/11) 91% | (3/3) 100% | (1/2) 50%
[src/App.tsx](../blob/664783e/src/App.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (0/0) NaN%
[src/index.tsx](../blob/664783e/src/index.tsx) | (3/3) 100% | (3/3) 100% | (0/0) NaN% | (0/0) NaN%
[src/../Authentication/index.ts](../blob/664783e/src/adapters/Authentication/index.ts) | (2/2) 100% | (2/2) 100% | (1/1) 100% | (0/0) NaN%
[src/adapters/Base/index.ts](../blob/664783e/src/adapters/Base/index.ts) | (8/8) 100% | (8/8) 100% | (2/2) 100% | (2/2) 100%
[src/adapters/BaseAuth/index.ts](../blob/664783e/src/adapters/BaseAuth/index.ts) | (10/10) 100% | (10/10) 100% | (4/4) 100% | (2/2) 100%
[src/adapters/Survey/index.ts](../blob/664783e/src/adapters/Survey/index.ts) | (1/1) 100% | (1/1) 100% | (1/1) 100% | (0/0) NaN%
[src/components/Alert/index.tsx](../blob/664783e/src/components/Alert/index.tsx) | (3/3) 100% | (3/3) 100% | (2/2) 100% | (0/0) NaN%
[src/../index.tsx](../blob/664783e/src/components/BackgroundImage/index.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (2/2) 100%
Other (21 more) | (85/85) 100% | (85/85) 100% | (31/31) 100% | (25/25) 100%
Total | (165/191) 86% | (165/191) 86% | (58/73) 79% | (43/54) 80%

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against aa1b364e197d9b559cbbb50786fa629029538c0d
</p>

File: src/components/Dashboard/Content/index.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ const DashboardContent = ({
   ) : (
     <div className="flex flex-col h-full" {...attributes}>
       <div
-        style={{ backgroundImage: `url(${surveys[currentPosition].imageUrl})` }}
+        style={{ backgroundImage: `url(${surveys[currentPosition].coverImageUrl})` }}
```
Comment (tyrro) [inline src/components/Dashboard/Content/index.tsx]: I think you can get a high-resolution image if you append the URL with `l`:

- low resolution: https://dhdbhh0jsld0o.cloudfront.net/m/1ea51560991bcb7d00d0_
- high resolution: https://dhdbhh0jsld0o.cloudfront.net/m/1ea51560991bcb7d00d0_l

we can write a helper to do this for us - we then can use the helper here 🙌🏿

**Review by tyrro — CHANGES_REQUESTED** [review]

File: src/components/Dashboard/Content/index.tsx (hunk 1 above)
Comment (manh-t) [reply src/components/Dashboard/Content/index.tsx]: @tyrro created in [c63ecf1](https://github.com/manh-t/react-survey/pull/55/commits/c63ecf129ba997d459fa126c73f9876f50275bb1)

File: src/components/Dashboard/Content/index.tsx [hunk 2]
```diff
@@ -48,8 +49,8 @@ const DashboardContent = ({
   ) : (
     <div className="flex flex-col h-full" data-test-id={dashboardDataTestIds.content}>
       <div
-        style={{ backgroundImage: `url(${surveys[currentPosition].imageUrl})` }}
-        className="w-full h-[302px] rounded-[12px] bg-cover duration-500 ease-in-out"
+        style={{ backgroundImage: `url(${getHighResolutionImage(surveys[currentPosition].coverImageUrl)})` }}
+        className={`w-full h-[302px] rounded-[12px] bg-cover duration-500 ease-in-out`}
```
Comment (hoangmirs) [inline src/components/Dashboard/Content/index.tsx]: Why do you need this change?

File: src/helpers/deserializer.ts [hunk 3]
```diff
@@ -6,6 +6,8 @@ import { JSONObject } from './json';
 
 export type DeserializableResponse = AxiosResponse<Deserializer>;
 
+export type DeserializableArrayResponse = AxiosResponse<Deserializer[]>;
```
Comment (hoangmirs) [inline src/helpers/deserializer.ts]: Should we use the `List` to make it consistent?
```suggestion
export type DeserializableListResponse = AxiosResponse<Deserializer[]>;
```

**Review by hoangmirs — APPROVED** [review]

File: src/helpers/image.test.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { getHighResolutionImage } from './image';
+
+describe('image helper', () => {
+  describe('given an image url', () => {
```
Comment (nvminhtue) [inline src/helpers/image.test.ts]:
```suggestion
describe('image helper', () => {
    describe('getHighResolutionImage', () => {
	  describe('given an image url', () => {
```

**Review by nvminhtue — APPROVED** [review]

File: src/components/Dashboard/Content/index.tsx (hunk 2 above)
Comment (manh-t) [reply src/components/Dashboard/Content/index.tsx]: @hoangmirs my bad, I forgot to change it back

**Review by tyrro — APPROVED** [review]

**Pull request merged**